pkg/urlutil/urlerrorhelper.go

```diff
@@ -0,0 +1,26 @@
+package urlutil
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func RemoveUrlFromErr(err error) error {
+	strSegments := strings.Split(err.Error(), " ")
+	for i, v := range strSegments {
+		if IsValidUrl(v) {
+			// we found a url
+			strSegments[i] = "[REDACTED]"
```

Comment:
I think there is still value in logging and including portions of the URL especially for diagnostic/debugging. Can you redact just the query parameters in the URL string?

Reply:
I cannot redact what I cannot parse. So, a malformed url, say,  http// could pass through to the error in the end. So I catch that possibility here. The error here is wrapped with failed to download file[%d], index, so we'll still know which of the provided urls failed.

Reply:
This behavior is consistent with other errors like getting a 404. But there error prints the index of the url provided that failed, so we can still know which one failed.